src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -0,0 +1,390 @@
+import com.minecolonies.coremod.util.InventoryUtils;
+import com.minecolonies.coremod.util.LanguageHandler;
+import com.minecolonies.coremod.util.Utils;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Class which handles the tileEntity of our colonyBuildings.
+ */
+public class TileEntityWareHouse extends TileEntityColonyBuilding
+{
+    /**
+     * List which contains the currentTasks to be executed by the deliveryman.
+     */
+    private final CopyOnWriteArrayList<AbstractBuilding> list = new CopyOnWriteArrayList<>();
+
+    /**
+     * Index of last controlled building.
+     */
+    private int index = 1;
+
+    /**
+     * Empty standard constructor.
+     */
+    public TileEntityWareHouse()
+    {
+        super();
+    }
+
+    @NotNull
+    @Override
+    public NBTTagCompound getUpdateTag()
+    {
+        return writeToNBT(new NBTTagCompound());
+    }
+
+    @Override
+    public void update()
+    {
+        super.update();
+
+        if(getColony() != null)
+        {
+            Map<BlockPos, AbstractBuilding> buildingMap = getColony().getBuildings();
+
+            if (buildingMap.size() < this.index)
+            {
+                this.index = 1;
+            }
+
+            int i = 1;
+            for (Map.Entry<BlockPos, AbstractBuilding> buildingEntry : buildingMap.entrySet())
+            {
+                if (i == index)
+                {
+                    if(buildingEntry.getValue() instanceof AbstractBuildingWorker
+                            && !list.contains(buildingEntry.getValue())
+                            && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                    {
+                        checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue());
+                    }
+                    this.index++;
+                }
+                i++;
+            }
+        }
+    }
+
+    /**
+     * Get the first task in the list.
+     * @return the building which needs a delivery.
+     */
+    @Nullable
+    public AbstractBuilding getTask()
+    {
+        if(list.isEmpty())
+        {
+            return null;
+        }
+        return list.remove(0);
+    }
+
+    /**
+     * Check if the required items by the building are in the wareHouse.
+     * @param buildingEntry the building requesting.
+     * @return true if has something in warehouse to deliver.
+     */
+    public boolean checkInWareHouse(final AbstractBuilding buildingEntry)
+    {
+        if(buildingEntry.areItemsNeeded())
+        {
+            for(ItemStack stack : buildingEntry.getNeededItems())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'stack' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)